walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,55 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.stream.*;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    //Approach: start form start of day, and add available timeranges as we traverse through the events' timeranges
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        Collection<TimeRange> validTimeRanges = new ArrayList();
+        long duration = request.getDuration();
+        int start = 0;
+        int end = 0;
+        int lastEnd = 0;
```

Comment:
consider using getTimeInMinutes() from the TimeRange file to initialize these variables. 
Alternatively, name the variables with a Minutes suffix: startMinutes, endMinutes, etc.

Reply:
Done!